tests/fixture/mod.rs

```diff
@@ -1,48 +1,44 @@
-use std::env;
+mod wrangler_toml;
+pub use wrangler_toml::{EnvConfig, KvConfig, SiteConfig, WranglerToml};
+
 use std::fs;
 use std::fs::File;
 use std::io::prelude::*;
+use std::mem::ManuallyDrop;
 use std::path::PathBuf;
-use std::process::Command;
-use std::sync::Mutex;
+use std::sync::MutexGuard;
+use std::thread;
 
-lazy_static! {
-    static ref BUILD_LOCK: Mutex<u8> = Mutex::new(0);
-}
+use tempfile::TempDir;
+use toml;
 
-const BUNDLE_OUT: &str = "./worker";
+const BUNDLE_OUT: &str = "worker";
 
 pub struct Fixture {
-    name: String,
+    // we wrap the fixture's tempdir in a `ManuallyDrop` so that if a test
+    // fails, its directory isn't deleted, and we have a chance to manually
+    // inspect its state and figure out what is going on.
+    dir: ManuallyDrop<TempDir>,
```

Comment:
Super, super nice! Does this mean testers will need to manually delete these leftover directories if there is a panic?

Reply:
yes they will not be cleaned up automatically; however since they are randomly generated by the OS, they will not cause collisions with subsequent test runs. should be trivial to write a shell script to clean them up after, as they consistently go to the same directory.